### PR TITLE
WIP HPCC-24691 Proposal for ESP logging compliance

### PR DESCRIPTION
**Pull request by rpastrana**
- Bare metal logging unaffected
-- Continue to support proc/@logLevel
-- Continue to support espcontrol::setloglevel
-- Filter logging output level via jlog constructs
-- Normalize Legacy esp log levels to jlog log detail range
-- Replace ESPLOG() w/ jlog XXXLOG macros
-- ESPLOG(ctx) to provide appropriate log message class
-- Remove dependance on legacy log levels

- Container mode
-- Global, or component logging standards via yaml
-- Remove dependence on esp specific loglevel config
-- espcontrol::setloglevel to update loghandler

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
WIP
@afishbeck per offline discussion

**Comment by rpastrana** [comment]
@AttilaVamos I'm trying to determine why this change would have caused the smoketest to fail, and the errors almost seem unrelated. Are these failures legitimate? Thanks.

**Comment by AttilaVamos** [comment]
@rpastrana Actually I don't know (yet) what is the problem, but you can reproduce it if you execute an ecl code with ecl tool like this:
```
ecl run -v -t hthor ~/HPCC-Platform/testing/regress/ecl/setup/roxie_echo.ecl
Using eclcc path /opt/HPCCSystems/bin/eclcc
EXEC: Creating PIPE program process : '/opt/HPCCSystems/bin/eclcc -E "/home/ati/HPCC-Platform/testing/regress/ecl/setup/roxie_echo.ecl"' - hasinput=0, hasoutput=1 stderrbufsize=0
EXEC: Pipe: process 23069 complete 0
http response content = <?xml version="1.0" encoding="utf-8"?><soap:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:wsse="http://schemas.xmlsoap.org/ws/2002/04/secext"><soap:Body><WUCheckFeaturesResponse xmlns="urn:hpccsystems:ws:wsworkunits"><BuildVersionMajor>7</BuildVersionMajor><BuildVersionMinor>10</BuildVersionMinor><BuildVersionPoint>21</BuildVersionPoint><maxRequestEntityLength>8000000</maxRequestEntityLength><Deployment><UseCompression>1</UseCompression></Deployment></WUCheckFeaturesResponse></soap:Body></soap:Envelope>

Deploying ECL Archive /home/ati/HPCC-Platform/testing/regress/ecl/setup/roxie_echo.ecl
http response content = <?xml version="1.0" encoding="utf-8"?><soap:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:wsse="http://schemas.xmlsoap.org/ws/2002/04/secext"><soap:Body><WUDeployWorkunitResponse xmlns="urn:hpccsystems:ws:wsworkunits"><Workunit><Wuid>W20200904-150634</Wuid><Owner></Owner><Cluster>hthor</Cluster><Jobname>roxie_echo</Jobname><StateID>1</StateID><State>compiled</State><Protected>0</Protected><DateTimeScheduled>1970-01-01T01:00:00+01:00</DateTimeScheduled><Snapshot></Snapshot><IsPausing>0</IsPausing><ThorLCR>0</ThorLCR><EventSchedule>0</EventSchedule><Query></Query><ErrorCount>0</ErrorCount><WarningCount>0</WarningCount><InfoCount>0</InfoCount><AlertCount>0</AlertCount></Workunit></WUDeployWorkunitResponse></soap:Body></soap:Envelope>

Deployed
   wuid: W20200904-150634
   state: compiled

Running deployed workunit W20200904-150634
http response content = <?xml version="1.0" encoding="utf-8"?><soap:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:wsse="http://schemas.xmlsoap.org/ws/2002/04/secext"><soap:Body><WURunResponse xmlns="urn:hpccsystems:ws:wsworkunits"><Wuid>W20200904-150634</Wuid><State>completed</State><Results>&lt;Result&gt;
&lt;Dataset name=&apos;Peeps&apos;&gt;
&lt;/Dataset&gt;
&lt;/Result&gt;</Results></WURunResponse></soap:Body></soap:Envelope>
<Result>
<Dataset name='Peeps'>
</Dataset>
</Result>
ati@ati-VirtualBox:~/Smoketest/PR-14134/HPCC-Platform/testing/regress$ 
```
I think the line starts with "http response content = "  shouldn't be there.

Without your changes the result looks like this:
```
ecl run -v -t hthor ~/HPCC-Platform/testing/regress/ecl/setup/roxie_echo.ecl
Using eclcc path /opt/HPCCSystems/bin/eclcc
EXEC: Creating PIPE program process : '/opt/HPCCSystems/bin/eclcc -E "/home/ati/HPCC-Platform/testing/regress/ecl/setup/roxie_echo.ecl"' - hasinput=0, hasoutput=1 stderrbufsize=0
EXEC: Pipe: process 23376 complete 0

Deploying ECL Archive /home/ati/HPCC-Platform/testing/regress/ecl/setup/roxie_echo.ecl

Deployed
   wuid: W20200904-150847
   state: compiled

Running deployed workunit W20200904-150847
<Result>
<Dataset name='Peeps'>
</Dataset>
</Result>
```

**Comment by rpastrana** [comment]
Thanks @AttilaVamos I'll look into it

**Comment by afishbeck** [comment]
@rpastrana the idea looks good.  Not sure exactly about the mapping or whether it covers everything we use level for.  maybe, @timothyklemm this is a work in progress, so focus on the idea not the details, but can you comment as well?

File: esp/platform/espcontext.cpp [hunk 1]
```diff
@@ -929,27 +975,40 @@ unsigned getSlowProcessingTime()
...
-    if (getEspLogLevel(NULL)>=level)
-    {
-        va_list args;
-        va_start(args,fmt);
-        VALOG(MCdebugInfo, unknownJob, fmt, args);
-        va_end(args);
-    }
+    va_list args;
+    va_start(args,fmt);
+    VALOG(getEspLogCategory(nullptr), unknownJob, fmt, args);
```
Comment (timothyklemm) [inline esp/platform/espcontext.cpp:991]: Is this correct? By ignoring the level passed to the function, aren't we effectively giving every message the same log level (i.e., the level assigned to the process)? If the process is configured with LogMin, then all messages will be logged if any message is logged.

Comment (rpastrana) [reply esp/platform/espcontext.cpp:991]: yep, thanks for pointing out, will restore, but ultimately ESPLOG will go away under this proposal (at least this version of it)

**Comment by richardkchapman** [comment]
This should be targeting master, not 7.10.x

**Comment by AttilaVamos** [comment]
@rpastrana The indexcachebug failure is not related to this PR. There is a JIRA to fix it. The schedule1 not related as well and I am sure after this PR will be re-schedule both error will be gone.

**Comment by rpastrana** [comment]
@afishbeck - we've talked about this one before but it's been lingering for a long time so I want to revisit. Long story short, this change migrates ESP specific logging to the more standard JLog mechanism. The PR contains the lower level changes needed, but a thorough migration of all ESPLog instances would have to be done in conjunction but I'm not sure if we have resources to perform that task. Let me know if you want to pursue this task to completion.

**Comment by afishbeck** [comment]
@rpastrana the idea of being consistent with other components is appealing, but I don't think we have the resources to really focus on this right now.  Perhaps we should revisit later?  Like once  users have migrated to the cloud.  On the other hand if anyone, including yourself, thinks its worth the effort right away, I could easily be convinced.

**Pull request closed without merging**